### PR TITLE
feat: allow passing an undefined theme to ThemeProvider

### PR DESCRIPTION
**Pull request by gbkr**
This allows dynamically passing in a theme - if it's undefined it just defaults to the Zopa theme

### CODE REVIEW

**Review by calvin-fung-zopa — APPROVED** [review]
LGTM!

File: src/components/styles/Theme.tsx [hunk 1]
```diff
@@ -285,16 +285,15 @@ export const zopaTheme: AppTheme = {
...
+export const useThemeContext = () => React.useContext(ThemeContext);
 
-export const ThemeProvider = ({ children, theme }: { children: React.ReactNode; theme: Partial<AppTheme> }) => {
-  const mergedTheme: AppTheme = { ...zopaTheme, ...theme };
+export const ThemeProvider = ({
+  children,
+  theme,
+}: {
+  children: React.ReactNode;
+  theme: Partial<AppTheme> | undefined;
```
Comment (admmasters) [inline src/components/styles/Theme.tsx:295]:
```suggestion
  theme?: Partial<AppTheme>;
```

what do you think?

Comment (admmasters) [reply src/components/styles/Theme.tsx:295]: Yeah I can see that kind of rationale for this api. Lets roll with it and then we can see how it works out in practice. Given we are the main consumers (its a closed api) - we can maybe revisit

**Review by admmasters — APPROVED** [review]

**Pull request merged**